src/main/python/vantiqsdk.py

```diff
@@ -1478,7 +1479,8 @@ async def subscribe(self, path: str, params: dict, callback: Callable[[str, dict
         if target_namespace is not None:
             request_id += '@' + target_namespace
 
-        if request_id in self.subscriptions.keys():
+        # Check to see if we already have a subscription for this path
+        if self.subscriptions.get(request_id, False):
```

Comment:
The previous code had a failure mode when the subscribe request failed with an exception and then was in a state that did not permit a retry of the subscription.